superset/db_engine_specs/base.py

```diff
@@ -901,6 +901,15 @@ def _extract_error_message(cls, ex: Exception) -> str:
         """Extract error message for queries"""
         return utils.error_msg_from_exception(ex)
 
+    @classmethod
+    def _reformat_error_message(cls, message: str) -> Tuple[str, str]:
+        """Reformat error message for user experience"""
+        splitted_message = re.split("\n+", message, 1)
+        if len(splitted_message) > 1:
+            return splitted_message[0], splitted_message[1]
+
+        return splitted_message[0], ""
```

Comment:
I would implement a no-op in `base.py`, something that does very little and maintains the current behavior as the default behavior. `big_query.py` would do the line parsing/splitting based on exactly how things are formatted there.